routers/repo/issue.go

```diff
@@ -207,16 +207,74 @@ func Issues(ctx *context.Context) {
 		}
 	}
 
-	// Get posters.
-	for i := range issues {
-		// Check read status
-		if !ctx.IsSigned {
-			issues[i].IsRead = true
-		} else if err = issues[i].GetIsRead(ctx.User.ID); err != nil {
-			ctx.Handle(500, "GetIsRead", err)
-			return
+	if !isPullList {
+		// Get posters.
+		for i := 0; i < len(issues); i++ {
+			// Check read status
+			if !ctx.IsSigned {
+				issues[i].IsRead = true
+			} else if err = issues[i].GetIsRead(ctx.User.ID); err != nil {
+				ctx.Handle(500, "GetIsRead", err)
+				return
+			}
 		}
+	} else {
+		var repoIDs = make([]int64, 0, len(issues))
+		var shas = make([]string, 0, len(issues))
+		var pullIDs = make([]int64, 0, len(issues))
```

Comment:
Yes, you are right. But I think store the last commit id in pullrequest table is the best way. I will send another PR to do that.

Reply:
If you do, call it `RepoSHA` instead of `CommitOption` since that better explains the purpose of the struct. Though in general I dislike `FunctionOptions` structs since they hide the arguments from the docs.

Reply:
@bkcsoft But 2 slices is not solution to that as you can't be sure about the order nor dependency between values on same key. I don't think that you are hiding something with structs in go.